src/routes/transactions/mappers/common/transactions.mapper.ts

```diff
@@ -0,0 +1,182 @@
+import { Injectable } from '@nestjs/common';
+import {
+  isEthereumTransaction,
+  isModuleTransaction,
+  isMultisigTransaction,
+  Transaction as TransactionDomain,
+} from '../../../../domain/safe/entities/transaction.entity';
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import { ModuleTransactionMapper } from '../module-transactions/module-transaction.mapper';
+import { MultisigTransactionMapper } from '../multisig-transactions/multisig-transaction.mapper';
+import { MultisigTransaction } from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { ModuleTransaction } from '../../../../domain/safe/entities/module-transaction.entity';
+import { EthereumTransaction } from '../../../../domain/safe/entities/ethereum-transaction.entity';
+import { groupBy } from 'lodash';
+import { IncomingTransferMapper } from '../transfers/transfer.mapper';
+import { TransactionItem } from '../../entities/transaction-item.entity';
+import { DateLabel } from '../../entities/date-label.entity';
+import { Transfer } from '../../../../domain/safe/entities/transfer.entity';
+
+class TransactionDomainGroup {
+  timestamp: number;
+  transactions:
+    | MultisigTransaction[]
+    | ModuleTransaction[]
+    | EthereumTransaction[];
+}
+
+@Injectable()
+export class TransactionsMapper {
+  constructor(
+    private readonly multisigTransactionMapper: MultisigTransactionMapper,
+    private readonly moduleTransactionMapper: ModuleTransactionMapper,
+    private readonly incomingTransferMapper: IncomingTransferMapper,
+  ) {}
+  async mapTransactions(
+    chainId: string,
+    transactionsDomain: TransactionDomain[],
+    safe: Safe,
+    offset: number,
+    timezoneOffset?: string,
+  ): Promise<[TransactionItem | DateLabel]> {
+    const prevPageTimestamp = this.getPreviousDayTimestamp(
+      transactionsDomain,
+      offset,
+      timezoneOffset,
+    )?.getTime();
+    // Remove first item that was requested to get previous day timestamp
+    if (prevPageTimestamp !== undefined) {
+      transactionsDomain = transactionsDomain.slice(1);
+    }
+    const transactionsDomainGroups = this.groupByDay(
+      transactionsDomain,
+      timezoneOffset,
+    );
+
+    const transactionList = await Promise.all(
+      transactionsDomainGroups.map(async (transactionGroup) => {
+        const transactions: (TransactionItem | DateLabel)[] = [];
+        if (transactionGroup.timestamp != prevPageTimestamp) {
+          transactions.push(new DateLabel(transactionGroup.timestamp));
+        }
+        transactions.push(
+          ...(await this.mapGroupTransactions(transactionGroup, chainId, safe)),
+        );
+        return await Promise.all(transactions.flat());
```

Comment:
I think all the promises should be resolved at this point (mappings are awaited in the line 63), so I think this could be simplified to `return transactions.flat()`

Reply:
Done 21343d26f082811dfac619628436595613a509eb